mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -5378,29 +5431,95 @@ pub struct HistoricalOrder {
     conf_settings: Option<OrderConfirmationsSettings>,
 }
 
-pub async fn orders_kick_start(ctx: &MmArc) -> Result<HashSet<String>, String> {
+/// Initializes and restarts the order matching system by loading saved orders from storage
+/// and populating the order contexts.
+///
+/// # Notes
+/// - For maker orders, only those with matching `rel_coin_account_id` are processed.
+/// - For taker orders, only those with matching `base_coin_account_id` are processed.
+/// - This ensures account consistency for both maker and taker orders.
+/// - Invalid orders are silently skipped and not added to the order contexts.
+pub async fn orders_kick_start(ctx: &MmArc, db_id: Option<&str>) -> Result<HashSet<String>, String> {
     let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(ctx));
-
     let storage = MyOrdersStorage::new(ctx.clone());
-    let saved_maker_orders = try_s!(storage.load_active_maker_orders().await);
-    let saved_taker_orders = try_s!(storage.load_active_taker_orders().await);
+    let saved_maker_orders = try_s!(storage.load_active_maker_orders(db_id).await);
+    let saved_taker_orders = try_s!(storage.load_active_taker_orders(db_id).await);
     let mut coins = HashSet::with_capacity((saved_maker_orders.len() * 2) + (saved_taker_orders.len() * 2));
 
     {
-        let mut maker_orders_ctx = ordermatch_ctx.maker_orders_ctx.lock();
         for order in saved_maker_orders {
-            coins.insert(order.base.clone());
-            coins.insert(order.rel.clone());
-            maker_orders_ctx.add_order(ctx.weak(), order.clone(), None);
+            let order_rel_coin_db_id = order.rel_coin_account_id.clone().unwrap_or(ctx.default_db_id());
+
+            loop {
```

Comment:
This loop and the other one will never end if the other coin is not enabled with the right address / `db_id`. We  should rethink this whole kickstarting approach. This is where the code looks for if the coin for an order is enabled or not https://github.com/KomodoPlatform/komodo-defi-framework/blob/117aa4402f6f8237ef95b56818b3f41e0c4ed987/mm2src/mm2_main/src/lp_ordermatch.rs#L4644-L4654

Reply:
fixed